groovy/gradle/src/org/netbeans/modules/gradle/api/GradleDependency.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.gradle.api;
+
+import java.io.File;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ *
+ * @author Laszlo Kishalmi
+ */
+public abstract class GradleDependency implements Serializable, Comparable<GradleDependency> {
+
+    public static enum Type {
+        UNRESOLVED, PROJECT, MODULE, FILE
+    }
+
+    final String id;
+
+    public GradleDependency(String id) {
```

Comment:
`protected` maybe be better. Or, if you only have fixed number of subclasses, make the constructor package private.

Reply:
Agreed.